### PR TITLE
#398: Update asciidoc-writers-guide.adoc

### PR DESCRIPTION
**Pull request by hemanik**
Added tip to include relative link to a file in a sub directory using xref inline macro.
issue:#398

### CODE REVIEW

File: docs/asciidoc-writers-guide.adoc [hunk 1]
```diff
@@ -366,6 +366,8 @@ include::{includedir}/xref.adoc[]
 
 include::{includedir}/xref-relative.adoc[]
 
+TIP: The xref inline macro by default resolves the link to HTML, to create a relative link to an asciidoc file in another subdirectory include `ifdef::env-github[:outfilesuffix: .adoc]` below the document header and drop the ".adoc" extension after the filename.
```
Comment (ecki) [inline docs/asciidoc-writers-guide.adoc]: Is there a method which is more neutral (support any output or source format). I canimagine there are more edit environents than GitHub?

Comment (hemanik) [inline docs/asciidoc-writers-guide.adoc]: @ecki, Thanks for pointing out, it shouldn't indeed be limited to just GitHub. 
I am not aware of any generic way, will explore and update the PR.

Comment (mojavelinux) [inline docs/asciidoc-writers-guide.adoc]: It's important to note that the xref macro only links to another file if the target ends in a hash path (e.g., `#` or `#anchor-name`). The `#` is how Asciidoctor distinguishes between an internal xref and an external xref.

> Is there a method which is more neutral (support any output or source format).

The `env-*` attributes are set by the integrations. For example, the code that runs on GitHub (and GitLab) sets the `env-github` attribute. You should expect that each integration has a similar attribute. For example, Awestruct sets `env-site`. We've tried to introduce generic attributes where possible, but in some cases like GitHub, the environment is so unique that it really only applies to that specific integration.

Btw, the browser previews also need the `outfilesuffix` to be `.adoc`. Both the Chrome and Firefox plugin set this value by default now. You can also use `env-browser` in those environments.

Comment (mojavelinux) [inline docs/asciidoc-writers-guide.adoc]: Eventually, we'll want to document all these environment attributes in the user manual. At the time of this writing, they are still evolving.

Comment (mojavelinux) [inline docs/asciidoc-writers-guide.adoc]: I'd like to see outfilesuffix set to `.adoc` by default on GitHub. There's an issue tracking that change. See https://github.com/github/markup/issues/519

**Comment by mojavelinux** [comment]
I incorporated this into https://github.com/asciidoctor/asciidoctor.org/commit/731a2878f40d976db9c2997bcf536034b1eb9da1.

This is now the default behavior on GitHub.

**Pull request closed without merging**